### PR TITLE
Ratakondala-arun/windows-support

### LINKED ISSUE #380: windows-support
https://docs.flutter.dev/deployment/windows#add-app-icons

### PR DESCRIPTION
**Pull request by RatakondalaArun**
- added support for windows (c22a950c2533c439f2d6fb190d01a991dfb2f071)
- added tests for windows (77b8ada537af2f66824bb8f9354640a244aeb79f)

closes #380

### CODE REVIEW

File: lib/windows/windows_icon_generator.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import 'package:image/image.dart';
+import 'package:path/path.dart' as path;
+
+import '../abs/icon_generator.dart';
+import '../constants.dart' as constants;
+import '../custom_exceptions.dart';
+import '../utils.dart' as utils;
```
Comment (MarkOSullivan94) [inline lib/windows/windows_icon_generator.dart:7]: Let's stick to not using relative imports

**Review by MarkOSullivan94 — CHANGES_REQUESTED** [review]
Left a couple comments other than that all great!

File: lib/windows/windows_icon_generator.dart (hunk 1 above)
Comment (RatakondalaArun) [reply lib/windows/windows_icon_generator.dart]: fixed in b421894

File: lib/constants.dart [hunk 2]
```diff
@@ -42,6 +42,20 @@ String webIndexFilePath = path.join(webDirPath, 'index.html');
...
+
+/// Relative path to windows resources directory
+String windowsResourcesDirPath = path.join(windowsDirPath, 'runner', 'resources');
+
+/// Relative path to windows icon file path
+String windowsIconFilePath = path.join(windowsResourcesDirPath, 'app_icon.ico');
+
+/// Default windows icon size for flutter
+///
+const int kWindowsIconSize = 48;
```
[the remark this answers is not in the document]
Comment (RatakondalaArun) [reply lib/constants.dart]: fixed in 65a5128

**Comment by RatakondalaArun** [comment]
@MarkOSullivan94 made all the changes you requested for

**Comment by RatakondalaArun** [comment]
> Let's stick to not using relative imports
> https://github.com/fluttercommunity/flutter_launcher_icons/pull/382#discussion_r929713721


we can use the `always_use_package_imports` linter rule to warn collaborators against using relative imports

I recommend we add this rule in separate PR since this will be out of the scope of this PR

**Review by MarkOSullivan94 — APPROVED** [review]
LGTM 👍

**Pull request merged**